packages/bitcore-node/src/providers/chain-state/svm/api/csp.ts

```diff
@@ -0,0 +1,407 @@
+import { CryptoRpc } from 'crypto-rpc';
+import { SolRpc } from 'crypto-rpc/lib/sol/SolRpc'
+import { instructionKeys } from 'crypto-rpc/lib/sol/transaction-parser';
+import { Readable } from 'stream';
+import Config from '../../../../config';
+import logger from '../../../../logger';
+import { CacheStorage } from '../../../../models/cache';
+import { Storage } from '../../../../services/storage';
+import { IBlock } from '../../../../types/Block';
+import { CoinListingJSON } from '../../../../types/Coin';
+import { IChainConfig, IProvider, ISVMNetworkConfig } from '../../../../types/Config';
+import { BroadcastTransactionParams, GetBalanceForAddressParams, GetBlockParams, GetCoinsForTxParams, GetEstimatePriorityFeeParams, GetWalletBalanceParams, IChainStateService, StreamAddressUtxosParams, StreamTransactionParams, StreamTransactionsParams, StreamWalletTransactionsParams } from '../../../../types/namespaces/ChainStateProvider';
+import {
+  getProvider,
+  isValidProviderType
+} from '../../external/providers/provider';
+import { ExternalApiStream } from '../../external/streams/apiStream';
+import { InternalStateProvider } from '../../internal/internal';
+
+export interface GetSolWeb3Response { rpc: SolRpc; connection: any; dataType: string; };
+
+export class BaseSVMStateProvider extends InternalStateProvider implements IChainStateService {
+  config: IChainConfig<ISVMNetworkConfig>;
+  static rpcs = {} as { [chain: string]: { [network: string]: GetSolWeb3Response[] } };
+
+  constructor(public chain: string = 'SOL') {
+    super(chain);
+    this.config = Config.chains[this.chain] as IChainConfig<ISVMNetworkConfig>;
+  }
+
+  async getRpc(network: string, params?: { type: IProvider['dataType'] }): Promise<GetSolWeb3Response> {
+    for (const rpc of BaseSVMStateProvider.rpcs[this.chain]?.[network] || []) {
+      if (!isValidProviderType(params?.type, rpc.dataType)) {
+        continue;
+      }
+
+      try {
+        await Promise.race([
+          rpc.rpc.getSlot({ commitment: 'confirmed' }).send(),
```

Comment:
This throws an error for me saying getSlot is not a function. `rpc.connection.getSlot` exists, though.